lisa/trace.py

```diff
@@ -1486,6 +1486,14 @@ class TxtTraceParser(TxtTraceParserBase):
                 'util': _KERNEL_DTYPE['util'],
             },
         ),
+        'sched_pelt_thermal': dict(
+            fields={
+                'cpu': _KERNEL_DTYPE['cpu'],
+                'load': _KERNEL_DTYPE['util'],
+                'rbl_load': _KERNEL_DTYPE['util'],
+                'util': _KERNEL_DTYPE['util'],
+            },
```

Comment:
The event has a `update_time` not listed here. That means the dataframe will not have this column. It's fine if it's somewhat useless for thermal signals (it's used to dump the PELT clock). 

Once you confirm that, it's good to merge

Reply:
'update_time' () could be a valuable information for sched_pelt_thermal but it's use case is very limited.
We would have the same issue for sched_pelt_[rt\|dl\|irq]. They also don't list the 'update_time' field but it appears in the trace event.

We have another issue with these trace events. Since we use the sched_pelt_rq_template for all of them, we plot `load`, `rbl_load` and `util` although only one has meaning to each of them and the other 2 are undefined.

rt, dl, irq: util_avg (load_avg & runnable_avg are meaningless)
thermal: load_avg (util_avg and runnable_avg are meaningless)

I would say we use sched_pelt_thermal w/o 'update_time' for now.

And then later we have to make a decision to whether:
(1) do we want to add 'update_time' to all of them?
(2) do we want to show only the valid load/rbl_load/util value for those 
    4 trace_events?
(I'm not sure why load_avg was chosen for thermal? Should be util_avg IMHO.

Reply:
> and the other 2 are undefined.
Undefined as in: the member of `struct sched_avg` is never initialized so we get arbitrary junk ? Or is it just that the field is set to some value but never updated by the kernel to something sensible ?

Either way, it sounds a bit strange that we end up attaching sched_avg structs to runqueues if only part of the data is used, and it can actually be harmful if some junk data become accessbile. AFAIU people usually try to keep structs as small as possible, but maybe it matters less for rq ? 

> I would say we use sched_pelt_thermal w/o 'update_time' for now.

Sounds good, we can always add it later if someone needs it.

If it turns out some values really don't make any sense, would you mind making a PR to split the events of each (i.e. not use a common template that forces the same format) and remove the corresponding senseless event fields in trace.py ?

Reply:
Yes, I can do this. I'll add patches to take care of (2). And I could also add 'update_time' to all, solving (1).